src/components/buttons/SearchButtonComponent.wc.svelte

```diff
@@ -104,7 +107,7 @@
 
             backend.send(
```

Comment:
To be honest, we should not call the backends here. I would rather move this to the App.svelte and remove it from the options. I think with a good documentation it will be better the current setup

Reply:
Yes, let's export a function `registerQueryCallback(callback: (query: AST) => ())` or similar. Then lens calls the callback with the query when the search button is clicked and the app executes the query in the callback and then calls `setSiteResult()` when it receives a result.

Reply:
Using an event as before as discussed